src/setup.rs

```diff
@@ -41,3 +45,14 @@ fn get_env_port() -> u16 {
 pub fn get_socket_addr() -> SocketAddrV4 {
     SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), get_env_port())
 }
+
+#[derive(Clone)]
+pub struct AppState {
+    connection_pool: Pool<Postgres>,
+}
+
+pub async fn create_app_state() -> AppState {
+    AppState {
+        connection_pool: database::get_connection_pool().await,
+    }
+}
```

Comment:
`connection_pool` is marked as unused by rust-analyzer. Was this state sharing tested on some endpoint?

Reply:
This is not resolved though, it's a legit thing. Unless we make this entire thing fall into the test endpoint category and make it work then.

Reply:
Well, we need endpoints to test resource sharing so, I suggest we test it once we have said endpoints in place.